Bottom: 3a21e5b10e751c4f8201672ee6e08cd1d53bc4e9
Top:    b6b6af350ef5ddee4831d021d6244cd17ef7e4bd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 20:27:23 +0100

Workaround crash where contour is passed as NULL to poly_PreContour






---

diff --git a/src/polygon.c b/src/polygon.c
index c697f3e..5c2d1f9 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -202,6 +202,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
